### ISSUE TITLE
Add support for multi-line single-line disable descriptions

### ISSUE THREAD

**Issue by nex3** [stylelint/stylelint#4886]
> Clearly describe the bug

With CSS-style multiline comments, it's possible to write disable descriptions (#4848) that span multiple lines:

```css
/* stylelint-disable declaration-no-important --
 * some long-winded description */
```

Unfortunately, this doesn't work with Sass-style single-line comments (when parsed as the SCSS syntax of course):

```scss
// stylelint-disable declaration-no-important --
// some long-winded description
```

> Which rule, if any, is the bug related to?

N/A

> What code is needed to reproduce the bug?

```css
a {
  // stylelint-disable color-no-invalid-hex --
  // foo
  color: #FF;
}
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
	"rules": {
		"color-no-invalid-hex": true
	}
}
```

> Which version of stylelint are you using?

Tested with 30bacfc2b6559b8428d44c01e612e343021e6be7

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint test.css --config scripts/visual-config.json`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to SCSS comments.

> What did you expect to happen?

I expected the description to be ignored and the lint to be disabled.

> What actually happened (e.g. what warnings or errors did you get)?

The lint was not disabled and still produced a warning

```shell
test.scss
 4:10  ✖  Unexpected invalid hex color "#FF"   color-no-invalid-hex
```

**Comment by hudochenkov** [comment]
This is probably due CSS comment being a single comment entity. While in Sass-style it's two separate comments.

**Comment by nex3** [comment]
It looks like you're right... if I add any text after the `--` it parses correctly, which suggests that it's trying to parse `stylelint-disable color-no-invalid-hex --` as a single entity and failing because it's looking for something like `-- .*` at the end.

As a user, though, I'd expect stylelint to treat multiple adjacent inline comments as a single unified comment (which is incidentally how they're exposed in Sass's own AST). For example, I'd also expect the following to disable the lint:

```scss
a {
  // stylelint-disable-next-line color-no-invalid-hex --
  // foo
  color: #FF;
}
```

Without support for this, #4868 would be of limited use in Sass files, since descriptions would have to be very short or risk overflowing line-length limits and producing readability problems.

**Comment by jeddy3** [comment]
> (which is incidentally how they're exposed in Sass's own AST)

That's very interesting. It would be great if postcss-scss matched this behaviour so that both the following are single comment nodes:

```css
/* stylelint-disable declaration-no-important --
 * some long-winded description */
```

```scss
// stylelint-disable declaration-no-important --
// some long-winded description
```

@nex3 Can you raise this issue upstream about this? If it can't be resolved upstream, we can work around it in stylelint.

**Comment by nex3** [comment]
Done: https://github.com/postcss/postcss-scss/issues/109

In the meantime, @jathak is working on a pull request to work around this representation issue in stylelint.

**Comment by jeddy3** [comment]
> @jathak is working on a pull request to work around this representation issue in stylelint.

Good stuff. It looks like [the change will be made upstream](https://github.com/postcss/postcss-scss/issues/109#issuecomment-671592794), but we can work around it in stylelint in the interim.
